indextools/pds4_index_tools.py

```
@@ -0,0 +1,344 @@
+    return bunprod_lid
+
+
+def get_collprod_filepath(collection_root, namespaces):
+    """Return the collection product filepath from the collection label file.
+
+    Inputs:
+        collection_root    The root of the collection label file.
+
+        namespaces         The namespaces present in the
+                           collection label file.
+
+    Returns:
+        collprod           The collection product file.
+    """
+    collection_file = collection_root.findall('pds:File_Area_Inventory',
+                                              namespaces=namespaces)
+    collprod = collection_file[0].File.file_name.text
+
+    return collprod
+
+
+def get_index_root(directory, path):
+    """Parse a data product label file and return its structured content.
+
+    Inputs:
+        directory     The path to the base directory.
+
+        path          The path to the data product file.
+
+    Returns:
+        index_root    The root element of the data product file.
+    """
+    index_root = (objectify.parse(os.path.join(directory, path),
+                                  objectify.makeparser(
+                                      remove_blank_text=True))
+                  .getroot())
+    return index_root
+
+
+def get_member_filepaths(directory, nlevels, keyword, subdirectory_name):
+    """Find and store all .xml/.lblx files whose filenames contain the keyword.
+
+    Inputs:
+        directory            The path to the base directory.
+
+        nlevels              The number of levels down within the directory the
+                             search can go.
+
+        keyword              The chosen keyword to search the directory with.
+
+        subdirectory_name    The name of the subdirectory containing the data
+                             products.
+
+    Returns:
+        files_found          The results of the file search. If empty, an
+                             exception is raised.
```

If you really want to return an exception instead of just an empty list, you should name which exception you raise.